File: Android/src/androidMain/kotlin/com/taskbridge/android/repository/impl/TaskTemplatesRepositoryImpl.kt
package com.taskbridge.android.repository.impl

import com.taskbridge.android.repository.TaskTemplatesRepository
import com.taskbridge.core.interactors.templates.TemplatesInteractor
import com.taskbridge.core.models.templates.TaskTemplatesState
import kotlinx.coroutines.flow.Flow

class TaskTemplatesRepositoryImpl(
    private val interactor: TemplatesInteractor
) : TaskTemplatesRepository {
    override val templatesState: Flow<TaskTemplatesState> = interactor.templatesState

    override suspend fun loadTemplates() {
        interactor.loadTemplates()
    }

    override suspend fun forceLoadTemplates() {
        interactor.forceLoadTemplates()
    }
}

File: iOS/TaskBridge/Repository/Impl/TaskTemplatesRepositoryImpl.swift
import Foundation
import TaskBridgeCore

@MainActor
class TaskTemplatesRepositoryImpl: TaskTemplatesRepository {
    private let interactor: TemplatesInteractor
    
    init(interactor: TemplatesInteractor) {
        self.interactor = interactor
    }
    
    var templatesState: AsyncStream<TaskTemplatesState> {
        return createStream(for: interactor.templatesState)
    }
    
    func loadTemplates() async throws {
        try await interactor.loadTemplates()
    }
    
    func forceLoadTemplates() async throws {
        try await interactor.forceLoadTemplates()
    }
}
